node/src/chain_specs/manta/mod.rs

```diff
@@ -0,0 +1,227 @@
+    GenesisConfig {
+        system: manta_runtime::SystemConfig {
+            code: manta_runtime::WASM_BINARY
+                .expect("WASM binary was not build, please build it!")
+                .to_vec(),
+        },
+        balances: manta_runtime::BalancesConfig {
+            balances: endowed_accounts[..endowed_accounts.len() / 2]
+                .iter()
+                .map(|k| {
+                    (
+                        k.clone(),
+                        10 * MANTA_ENDOWMENT / ((endowed_accounts.len() / 2) as Balance),
+                    )
+                })
+                .collect(),
+        },
+        // no need to pass anything to aura, in fact it will panic if we do. Session will take care
+        // of this.
+        aura: Default::default(),
+        sudo: manta_runtime::SudoConfig {
+            key: Some(root_key),
+        },
+        parachain_staking: ParachainStakingConfig {
+            candidates: invulnerables
+                .iter()
+                .cloned()
+                .map(|(account, _)| (account, NORMAL_COLLATOR_MINIMUM_STAKE))
+                .collect(),
+            delegations,
+            inflation_config: manta_runtime::staking::inflation_config::<manta_runtime::Runtime>(),
+        },
+        parachain_info: manta_runtime::ParachainInfoConfig {
+            parachain_id: MANTA_PARACHAIN_ID.into(),
+        },
+        collator_selection: manta_runtime::CollatorSelectionConfig {
+            invulnerables: invulnerables.iter().cloned().map(|(acc, _)| acc).collect(),
+            candidacy_bond: 10_000 * MANTA, // How many tokens will be reserved as collator
+            ..Default::default()
+        },
+        session: manta_runtime::SessionConfig {
+            keys: invulnerables
+                .iter()
+                .cloned()
+                .map(|(acc, session_keys)| {
+                    (
+                        acc.clone(),  // account id
+                        acc,          // validator id
+                        session_keys, // collator session keys
+                    )
+                })
+                .collect(),
+        },
+        parachain_system: Default::default(),
+        polkadot_xcm: PolkadotXcmConfig {
+            safe_xcm_version: Some(SAFE_XCM_VERSION),
+        },
+    }
+}
+
+/// Returns the Manta testnet chainspec.
+pub fn manta_testnet_config() -> Result<MantaChainSpec, String> {
+    let mut spec = MantaChainSpec::from_json_bytes(
+        &include_bytes!("../../../../genesis/manta-testnet-genesis.json")[..],
+    )?;
+    spec.extensions_mut().para_id = MANTA_PARACHAIN_ID;
+    Ok(spec)
+}
+
+/// Returns the Manta mainnet chainspec, generated to the below json file using
+/// using Ok(mainnet_genesis::manta_mainnet_config())
+/// cargo run -- build-spec --chain manta --raw
+pub fn manta_config() -> Result<MantaChainSpec, String> {
+    MantaChainSpec::from_json_bytes(&include_bytes!("../../../../genesis/manta-genesis.json")[..])
```

Comment:
one question here, why are we not using the raw json file for each of these @ghzlatarev ?
Seems that would be more resilient against client type changes.
```suggestion
    MantaChainSpec::from_json_bytes(&include_bytes!("../../../../genesis/manta-genesis-raw.json")[..])
```

Reply:
nvm, answering myself: if the input is raw, one can't do `manta build-spec --chain=manta`, the output will be identical to `manta build-spec --chain=manta --raw`